### PR TITLE
Improve TypeScript definitions & add test

### PR DESCRIPTION
**Pull request by VincentBel**

### CODE REVIEW

File: src/diff.js [hunk 1]
```diff
@@ -56,15 +56,15 @@ class MonacoDiffEditor extends React.Component {
 
   editorDidMount(editor) {
     this.props.editorDidMount(editor, monaco);
-    editor.onDidUpdateDiff((event) => {
+    editor.onDidUpdateDiff(() => {
```
Comment (VincentBel) [inline src/diff.js]: According to the type(enter <https://unpkg.com/monaco-editor@0.12.0/esm/vs/editor/editor.api.d.ts> and search `onDidUpdateDiff`), the callback function has no `event` parameter.

Comment (domoritz) [inline src/diff.js]: Can you explain this change? In https://github.com/superRaytin/react-monaco-editor/blob/master/src/editor.js#L62, we still pass event.

File: src/index.d.ts [hunk 2]
```diff
@@ -1,75 +1,63 @@
...
-  value: string | null;
-  defaultValue: string;
-  language: string;
-  theme: string | null;
-  options: object;
-  editorDidMount: (editor, monaco) => void;
-  editorWillMount: (monaco) => void;
-  onChange: (value, event) => void;
-  requireConfig: object;
-  context: object;
```
Comment (VincentBel) [inline src/index.d.ts]: It seems there are no `requireConfig` and `context` props.

File: src/diff.js (hunk 1 above)
Comment (VincentBel) [reply src/diff.js]: According to the type(enter <https://unpkg.com/monaco-editor@0.12.0/esm/vs/editor/editor.api.d.ts> and search `onDidUpdateDiff`), the callback function has no `event` parameter.

Comment (VincentBel) [reply src/diff.js]: And in the vscode repo:
https://github.com/Microsoft/vscode/blob/292732cd19cd4c6b0e11e779d14be480ff186ca8/src/vs/editor/browser/editorBrowser.ts#L776

**Comment by domoritz** [comment]
Thanks!

**Pull request merged**